Libs/MRML/Core/vtkMRMLSubjectHierarchyNode.cxx

```diff
@@ -1309,6 +1339,25 @@ std::string vtkSubjectHierarchyItem::GetUID(std::string uidName)
   return std::string();
 }
 
+//---------------------------------------------------------------------------
+std::vector<std::string> vtkSubjectHierarchyItem::GetUIDNames()
+{
+  std::vector<std::string> uidNameList;
+  std::map<std::string, std::string>::iterator uidIt;
+  for (uidIt=this->UIDs.begin(); uidIt!=this->UIDs.end(); ++uidIt)
+  {
+    uidNameList.push_back(uidIt->first);
+  }
+  return uidNameList;
+}
+
+//---------------------------------------------------------------------------
+bool vtkSubjectHierarchyItem::HasUID(std::string uidName)
+{
+  return (this->UIDs.size() > 0
```

Comment:
`this->UIDs.size() > 0` this checking looks redundant, find will always return UIDS.end() if UIDs is empty and it does not provide performance optimization as well.